### PR TITLE
feat(react): #MAG-424 implement folder navigation

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
folders and boards updated when navigating module
handle public boards and deleted boards/folders

## Checklist tests
navigate files in module

## Issue ticket number and link
[MAG-424](https://jira.support-ent.fr/browse/MAG-424)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/content-page/ContentPage.tsx [hunk 1]
```diff
@@ -1,26 +0,0 @@
-import React from "react";
```
Comment (alicedraillard) [inline frontend/src/components/content-page/ContentPage.tsx:1]: remove component that is not useful (reviewed with @guillaume-roux)

**Review by niko-amosse — APPROVED** [review]

File: frontend/src/components/board-list/BoardList.tsx [hunk 2]
```diff
@@ -1,103 +1,109 @@
...
+                        ...springs
+                        }}
+                    >
+                        <Card
+                            app={currentApp!}
+                            options={{
+                                type: "board",
+                                title: "",
+                            }}
+                            // onClick={() => {setIsToasterOpen()}}
```
Comment (FlorentMr) [inline frontend/src/components/board-list/BoardList.tsx:93]: commentaire

File: frontend/src/components/create-tab/createTab.tsx [hunk 3]
```diff
@@ -76,6 +76,7 @@ export const CreateTab: FunctionComponent<props> = ({
     board.displayNbFavorites = isFavoriteChecked;
     board.tags = tags;
 
+    createBoard(board.toJSON());
```
Comment (FlorentMr) [inline frontend/src/components/create-tab/createTab.tsx:79]: Pourquoi cette ligne est doublée?

File: frontend/src/components/folder-list/FolderList.tsx [hunk 4]
```diff
@@ -1,41 +1,64 @@
...
+}
 
-  const {
-    data: myFoldersResult,
-    isLoading: getFoldersLoading,
-    error: getFoldersError,
-  } = useGetFoldersQuery(false);
+export const FolderList: React.FunctionComponent<FolderListProps> = ({ currentFolder, onSelect }) => {
+    const { currentApp } = useOdeClient();
+    // const [isToasterOpen, setIsToasterOpen] = useToaster();
```
Comment (FlorentMr) [inline frontend/src/components/folder-list/FolderList.tsx:21]: commentaire

File: frontend/src/components/folder-list/FolderList.tsx [hunk 5]
```diff
@@ -1,41 +1,64 @@
...
   const springs = useSpring({
     from: { opacity: 0 },
     to: { opacity: 1 },
   });
 
+  useEffect(() => {
+    setFoldersQuery(currentFolder.id == FOLDER_TYPE.DELETED_BOARDS || !!currentFolder.deleted);
+  }, [currentFolder]);
+
+
```
Comment (FlorentMr) [inline frontend/src/components/folder-list/FolderList.tsx:60]: sauts de lignes en trop (faudra juste penser à linter + prettier)

File: frontend/src/models/folder.model.ts [hunk 6]
```diff
@@ -78,6 +86,23 @@ export class Folder {
     this._shared = value;
   }
 
+  get deleted(): any[] {
```
Comment (FlorentMr) [inline frontend/src/models/folder.model.ts:89]: possible de typer ce any et ceux d'après?

File: frontend/src/models/folder.model.ts [hunk 7]
```diff
@@ -87,10 +112,11 @@ export class Folder {
     const finalFolder: Folder = new Folder();
     finalFolder.id = navItem.id;
     finalFolder.title = navItem.name;
-    finalFolder.parentId = navItem.parentId;
+    // finalFolder.parentId = navItem.parentId;
```
Comment (FlorentMr) [inline frontend/src/models/folder.model.ts:115]: commentaire

File: frontend/src/components/board-list/BoardList.tsx (hunk 2 above)
Comment (alicedraillard) [reply frontend/src/components/board-list/BoardList.tsx:93]: laissé en vue de la MAG-355

File: frontend/src/components/folder-list/FolderList.tsx (hunk 4 above)
Comment (alicedraillard) [reply frontend/src/components/folder-list/FolderList.tsx:21]: idem MAG-355

File: frontend/src/components/folder-list/FolderList.tsx (hunk 5 above)
Comment (alicedraillard) [reply frontend/src/components/folder-list/FolderList.tsx:60]: linter + prettier seront appliqués à la fin ;)

**Review by FlorentMr — APPROVED** [review]

**Pull request merged**